internal/handlers/pg/msg_renamecollection.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pg
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgRenameCollection implements HandlerInterface.
+func (h *Handler) MsgRenameCollection(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	var err error
+	dbPool, err := h.DBPool(ctx)
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.Unimplemented(document, "writeConcern", "comment", "dropTarget"); err != nil {
+		return nil, err
+	}
+
+	sourceNamespace, err := common.GetRequiredParam[string](document, document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	sourceDB, collection, err := extractFromNamespace(sourceNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	targetNamespace, err := common.GetRequiredParam[string](document, "to")
+	if err != nil {
+		return nil, err
+	}
+
+	// we assume we cannot move a collection between databases, yet.
+	_, targetCollection, err := extractFromNamespace(targetNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	var qp pgdb.QueryParams
+
+	if qp.DB, err = common.GetRequiredParam[string](document, "$db"); err != nil {
+		return nil, err
+	}
+
+	if qp.DB != "admin" {
+		return nil, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrUnauthorized,
+			"renameCollection may only be run against the admin database.",
```

Comment:
Why? Is it a temporary thing or is it how MongoDB works?

Reply:
This is how MongoDB works.

Apparently the command version requires it to be run against the [admin database](https://www.mongodb.com/docs/manual/reference/command/renameCollection/#mongodb-dbcommand-dbcmd.renameCollection).

Reply:
@b1ron please add a test that shows an attempt to rename a collection in a non-admin database returns an error. I tried calling it in any database with mongosh, and it works.